apps/mull-api/src/app/user/user.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../entity';
+import { CreateUserInput, UpdateUserInput } from './inputs/user.input';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>
+  ) {}
+
+  findAll(): Promise<User[]> {
+    return this.userRepository.find();
+  }
+
+  findOne(id: number): Promise<User> {
+    return this.userRepository.findOne(id);
+  }
+
+  async findAllFriends(id: number): Promise<User[]> {
+    const { friends } = await this.userRepository.findOne(id, { relations: ['friends'] });
+    return friends;
+  }
+
+  async create(userInput: CreateUserInput): Promise<User> {
+    const { age, name, email, password } = userInput;
+    const newUser = this.userRepository.create();
+    newUser.name = name;
+    newUser.email = email;
+    newUser.password = password;
+    newUser.age = age;
```

Comment:
To make this easier to read you can pass in the entity properties as an object. Like below. Note that since the property name matches the value name we don't need `email: email` we can just write `email`

`const newUser = this.userRepository.create({
      name,
      email,
      password,
      age
    });`

Reply:
Good Idea @r-saba! That looks a lot cleaner. 🧹
As for testing, Nx generates test files next to the source files and I think we should just stick to what it provides us. 🗃

Reply:
Do you guys, encrypt the password before saving it to the database?

Reply:
Not yet, but that would be done for user registration. 🚀

Reply:
```
  create(userInput: CreateUserInput): Promise<User> {
    return this.userRepository.save(userInput);
  }
```
should work too.